presto-main/src/main/java/com/facebook/presto/operator/DynamicFilterSourceOperator.java

```diff
@@ -170,8 +186,13 @@ private DynamicFilterSourceOperator(
 
         this.blockBuilders = new BlockBuilder[channels.size()];
         this.valueSets = new TypedSet[channels.size()];
+        ImmutableList.Builder<Integer> minMaxChannelsBuilder = ImmutableList.builder();
         for (int channelIndex = 0; channelIndex < channels.size(); ++channelIndex) {
             Type type = channels.get(channelIndex).getType();
+            // Skipping DOUBLE and REAL in collectMinMaxValues to avoid dealing with NaN values
```

Comment:
Shall we support Double and Real?  What's the obstacle of dealing with NaN values? is it just because NaN is not comparable?

Reply:
> is it just because NaN is not comparable?

Yes - `NaN` breaks the ability to compare floating point values.
In addition, not many joins are using `REAL` / `DOUBLE` columns as keys.